src/crimson/osd/ops_executer.h

```diff
@@ -450,6 +450,23 @@ class OpsExecuter {
 
   version_t get_last_user_version() const;
 
+  bool has_cloning_ctx() const {
+    return cloning_ctx != nullptr;
+  }
+
+  const hobject_t& get_cloning_coid() const {
+    return cloning_ctx->coid;
```

Comment:
we should assert has_cloning_ctx to make this interface safe to use by any future user.